include/onnxruntime/core/framework/tensor.h

```diff
@@ -216,7 +216,7 @@ class Tensor final {
   TensorShape shape_;
   const PrimitiveDataTypeBase* dtype_;
   OrtMemoryInfo alloc_info_;
-  int64_t byte_offset_;
+  size_t byte_offset_;
```

Comment:
As it is for offset pointers, it can't be larger than the max value of size_t

Reply:
Would ptrdiff_t be better if it's a pointer delta?

Reply:
Good idea